src/libstd/sys/windows/backtrace.rs

```diff
@@ -157,3 +161,43 @@ unsafe fn _write(w: &mut Write) -> io::Result<()> {
 
     Ok(())
 }
+
+fn query_full_process_image_name() -> io::Result<PathBuf> {
+    unsafe {
+        let process_handle = Handle::new(c::OpenProcess(c::PROCESS_QUERY_INFORMATION,
+                                                        c::FALSE,
+                                                        c::GetCurrentProcessId()));
+        super::fill_utf16_buf(|buf, mut sz| {
+            c::QueryFullProcessImageNameW(process_handle.raw(), 0, buf, &mut sz); sz
```

Comment:
Reading `fill_utf16_buf` carefully and seeing that `QueryFullProcessImageNameW` will leave `sz` as is on error. You are right.